pkg/configuration/configuration.go

```diff
@@ -55,22 +55,41 @@ type Configuration interface {
 	PersistInStorage(key string)
 	SetStorage(storage Storage)
 	GetStorage() Storage
+
+	AutomaticEnv()
+	SetAutomaticEnv(b bool)
```

Comment:
Suggestion: I think the setter is a bit problematic as it creates the possibility for ambiguity, since you can set the parameter to false, which will have no effect. Better remove the setting and only have the getter `GetAutomaticEnv()` and `AutomaticEnv()`